### PR TITLE
Migrate NotebookRender to TypeScript

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/nteract/nteract/issues/3576.

Documenting this was a bit challenging because it isn't a siloed component so I felt weird about providing full-interactive examples in what should be an API reference. Not sure how to approach documenting these types of components yet.

### CODE REVIEW

**Comment by captainsafia** [comment]
Looks like I need to install some type declarations and write a few up. I'll work on this tomorrow.

**Comment by rgbkrk** [comment]
As we talked about in chat, I don't think we should document monolith components like this in our style guide. The style guide should live as a standalone section for us and other contributors to reference when building apps. This belongs in docs, though likely in a section devoted to blogging with notebooks, in this case using Gatsby.

File: packages/notebook-render/src/index.tsx [hunk 1]
```diff
@@ -29,15 +28,15 @@ import {
...
 
-type Props = {
-  displayOrder: Array<string>,
-  notebook: any,
-  transforms: Object,
-  theme: "light" | "dark"
+interface Props  {
+  displayOrder: Array<string>;
+  notebook: any;
+  transforms: Object;
```
Comment (charlesjolley) [inline packages/notebook-render/src/index.tsx]: Should be object (lowercase o)

File: packages/notebook-render/src/index.tsx [hunk 2]
```diff
@@ -29,15 +28,15 @@ import {
   Cells
 } from "@nteract/presentational-components";
 
-type Props = {
-  displayOrder: Array<string>,
-  notebook: any,
-  transforms: Object,
-  theme: "light" | "dark"
+interface Props  {
+  displayOrder: Array<string>;
```
Comment (charlesjolley) [inline packages/notebook-render/src/index.tsx]: Should be string [] per our earlier discussion? (Simple srrays use bracket. Complex types get array<>)

File: packages/notebook-render/src/index.tsx [hunk 3]
```diff
@@ -29,15 +28,15 @@ import {
...
+  notebook: any;
+  transforms: Object;
+  theme: "light" | "dark";
 };
 
-type State = {
-  notebook: any
+interface State {
+  notebook: any;
```
Comment (charlesjolley) [inline packages/notebook-render/src/index.tsx]: Can/should this be made more specific? You lose typing basically

**Review by charlesjolley — COMMENTED** [review]
Suggested a few new type colors for the bikeshed

File: packages/notebook-render/src/index.tsx (hunk 3 above)
Comment (rgbkrk) [reply packages/notebook-render/src/index.tsx]: Now that commutable is completely typescripted with some nice records, we could probably use the definition there. This component probably takes the raw notebook format or commutable format as props and stores the derived state as a commutable notebook. Both those types are in commutable.

File: types/nteract__presentational-components/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    export const Input: any;
+    export const Outputs: any;
+    export const Pagers: any;
+    export const Prompt: any;
+    export const PromptBuffer: any;
+    export const Source: any;
+    export const Cell: any;
+    export const Cells: any;
+    export const HeaderEditor: any;
+}
```
Comment (rgbkrk) [inline types/nteract__presentational-components/index.d.ts:12]: There's a part of me that wants to convert these to typescript next.

Comment (captainsafia) [reply types/nteract__presentational-components/index.d.ts:12]: Agreed! I mentioned this to @jdetle and I think he's working on it at the moment.

File: packages/notebook-render/src/index.tsx (hunk 3 above)
Comment (captainsafia) [reply packages/notebook-render/src/index.tsx]: Done. I used `Notebook | ImmutableNotebook`.

File: types/nteract__presentational-components/index.d.ts (hunk 4 above)
Comment (jdetle) [reply types/nteract__presentational-components/index.d.ts:12]: Yeah, on my todolist for today.

**Comment by captainsafia** [comment]
OK! I think this is good to merge. Took a while for me to get the tests passing with all the new types.

**Comment by rgbkrk** [comment]
Awesome, thank you

**Pull request merged**